### ISSUE TITLE
LD2410 doesn't send data

### ISSUE THREAD

**Issue by ady1972** [ApolloAutomation/MSR-2#11]
Hello i'm using your yaml-factory file to create a DIY version of your sensor, but LD2410 doesn't send any data to ESP32 it results always unkwown.
I have connected rx and tx pin of ld2410 to tx and rx pin pf ESP32-C3 devboard.
I have tested LD2410 with HLK Radar tool app and it works well.
What's the peoblem?

Thanks a lot

**Comment by TrevorSchirmer** [comment]
Hi,

We have RX and TX pins set to this

uart:
  tx_pin: GPIO21
  rx_pin: GPIO20

You'll want to double check yours are the same, it is easy to get them mixed up.


Are you powering it with 3V or 5V?

**Comment by ady1972** [comment]
Thanks for your answer.
I have checked pin GPIO21 and GPIO20 trying several times also to inert pin.
LD2410 is powered with 5v.

If everything is ok maybe I'll try another ESP32 PCB.

Thanks a lot for your answer

Ciao

Andrea


Il ven 13 set 2024, 00:42 Trevor Schirmer ***@***.***> ha
scritto:

**Comment by ady1972** [comment]
Tried another ESP32 borad but nothing to do. GPIO20 and 21 are RX and TX pin on ESP32 board (looking at wiring scheme)...so it seems that connections are ok.
Any suggestion ?

**Comment by TrevorSchirmer** [comment]
Unfortunately I'm not sure what else it could be if you've double checked the wiring and tried swapping the TX and RX to make sure it isn't backwards. Can you post your full YAML file?

**Comment by ady1972** [comment]
i have followed your guide :
downloaded bin file (apollo-msr-2-esp32c3.factory.bin) from Github and flashed using Web.esphome.io. I've also tried your web install link (https://apolloautomation.github.io/MSR-2/). Then connect to MSR-2 hotspot, choose my wifi , connect to it.
Then adopt it in ESPHOME addon of Home Assistant an finally configure device inside ESPHome section of home assistant.
Here below resulted yaml configuration

substitutions:
  name: apollo-msr-2-bef3e8
  friendly_name: Apollo MSR-2
packages:
  ApolloAutomation.MSR-2: github://ApolloAutomation/MSR-2/Integrations/ESPHome/MSR-2.yaml
esphome:
  name: ${name}
  name_add_mac_suffix: false
  friendly_name: ${friendly_name}
api:
  encryption:
    key: 4s06p3SAJWE/wuSgHoNji6SBj9dtKwiAhUYbmJANxuY=
wifi:
  ssid: !secret wifi_ssid
  password: !secret wifi_password

![IMG_20240914_083149](https://github.com/user-attachments/assets/f916de5b-330c-445e-985d-e430408d52d6)

I have also followed this procedure for MTR-1 version and it works perfctly.

Thanks a lot for your help.

Have a nice day
